### PR TITLE
Decorate more functionality with @doc and flesh out jsdocs for array ops.

### PR DESCRIPTION
**Pull request by nsthorat**
This PR:
* Adds more @docs for training / tracking.
* Moves floating methods to parts of static classes, and export sugar floating methods that alias them. This is so we can add decorators.
* Adds end line numbers to github code search URLs.
* Sort methods by name within a subheading in API reference.
* Renames randNormal => randomNormal, and randUniform => randomUniform to mirror TF: https://www.tensorflow.org/api_docs/python/tf/random_normal

This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/641)

### CODE REVIEW

**Comment by dsmilkov** [comment]
:lgtm_strong: Very cool!

---

Reviewed 33 of 33 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[src/math/array_ops.ts, line 28 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YbmmjP1D_53eLLpZG:-L4YbmmjP1D_53eLLpZH:b-bm30c2) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L28)):*
> ```XML
> export class Ops {
>   /**
>    * Creates a tensor with all elements set to 1.
> ```

Here and elsewhere: seems a little repetitive to say both `creates a tensor ...` and `This operation returns a tensor ...` Maybe just keep:
`Creates a tensor with all elements set to 1`. No need to say of dtype `dtype` and shape `shape` since the params are obvious and they have their own param docs.

---

*[src/math/array_ops.ts, line 32 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YbAHqs7G76heSc4_-:-L4YbAHqs7G76heSc4_0:b-lpk2sd) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L32)):*
> A array

typo: An array

---

*[src/math/array_ops.ts, line 117 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YbcQ9eNBQstomqH2T:-L4YbcQ9eNBQstomqH2U:b-9oi1oe) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L117)):*
> ```XML
>    * Creates a tensor with values sampled from a normal distribution.
>    *
>    * @param shape A array of integers defining the output tensor shape.
> ```

s/A array/An array/ here and elsewhere

---

*[src/math/array_ops.ts, line 259 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YcRP70nx2_5Jcqd4J:-L4YcRP70nx2_5Jcqd4K:bw0f3ta) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L259)):*
> ```XML
> 
>   /**
>    * Creates a one-hot array. The locations represented by `indices` take
> ```

Creates a one-hot tensor.

---

*[src/math/array_ops.ts, line 377 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YcLPBKRMAjS0jhPyU:-L4YcLPBKRMAjS0jhPyV:b-80tupl) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L377)):*
> ```XML
>    * Gather slices from array `x`'s axis `axis` according to `indices`
>    *
>    * @param x The array to transpose.
> ```

The tensor to transpose

---

*[src/math/array_ops_test.ts, line 23 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YcbzdIExeSCU-Y5E2:-L4YcbzdIExeSCU-Y5E3:b-3b0crb) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops_test.ts#L23)):*
> ```XML
> import * as util from '../util';
> 
> import {Scalar, Tensor, Tensor1D, Tensor2D, Tensor3D, Tensor4D} from './tensor';
> ```

if not too much trouble, would be great to go all in with dl.* and remove this import

---

*[src/math/backends/gradients.ts, line 164 at r1](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-L4YdU26VPN6u-UD2fO2:-L4YdU26VPN6u-UD2fO3:bhs3aaw) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/backends/gradients.ts#L164)):*
> ```XML
> }
> 
> export const gradientsScope = Gradients.gradientsScope;
> ```

Just FYI, no action needed: we don't expose this in index.ts since it's unclear if users will need it, especially with the new grad api `gradients(y)(x)`.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/pair-code/deeplearnjs/641#-:-L4Yf36d2v8OEmhGQKyH:bt7f1ep)*

**Comment by nsthorat** [comment]
Review status: 27 of 37 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[src/math/array_ops.ts, line 28 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YbmmjP1D_53eLLpZG:-L4Yfxlp0x5AwvCImsGg:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L28)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

Here and elsewhere: seems a little repetitive to say both `creates a tensor ...` and `This operation returns a tensor ...` Maybe just keep:
`Creates a tensor with all elements set to 1`. No need to say of dtype `dtype` and shape `shape` since the params are obvious and they have their own param docs.


</blockquote></details>

Done.

---

*[src/math/array_ops.ts, line 32 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YbAHqs7G76heSc4_-:-L4Yfwdp6VtwYRvnMcUW:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L32)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

> A array

typo: An array
</blockquote></details>

Done.

---

*[src/math/array_ops.ts, line 117 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YbcQ9eNBQstomqH2T:-L4Yg6-z5uKHW5D8g3eV:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L117)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

s/A array/An array/ here and elsewhere
</blockquote></details>

Done.

---

*[src/math/array_ops.ts, line 259 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YcRP70nx2_5Jcqd4J:-L4Yg9C52ioTxF13Zo3d:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L259)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

Creates a one-hot tensor.
</blockquote></details>

Done.

---

*[src/math/array_ops.ts, line 377 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YcLPBKRMAjS0jhPyU:-L4Yhw8gEm37xJZcvKwe:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops.ts#L377)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

The tensor to transpose
</blockquote></details>

Done.

---

*[src/math/array_ops_test.ts, line 23 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YcbzdIExeSCU-Y5E2:-L4Yi6Tt0RLCKQJhnZy0:b-896fix) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/array_ops_test.ts#L23)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

if not too much trouble, would be great to go all in with dl.* and remove this import
</blockquote></details>

Done.

---

*[src/math/backends/gradients.ts, line 164 at r1](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641#-L4YdU26VPN6u-UD2fO2:-L4Yi7aaEnutjD6xHSy3:brj023s) ([raw file](https://github.com/pair-code/deeplearnjs/blob/f045c54fcfa349abc1560c2de3e0fce8846eec5b/src/math/backends/gradients.ts#L164)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

Just FYI, no action needed: we don't expose this in index.ts since it's unclear if users will need it, especially with the new grad api `gradients(y)(x)`.
</blockquote></details>

Yep, this doesn't export anything, just wraps it in a class of static methods.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pair-code/deeplearnjs/641)*

**Comment by nsthorat** [comment]
Confirming CLA.
